pkg/manifest/ostree_deployment.go

```diff
@@ -204,13 +194,10 @@ func (p *OSTreeDeployment) serialize() osbuild.Pipeline {
 	kernelOpts := osbuild.GenImageKernelOptions(p.PartitionTable)
 	kernelOpts = append(kernelOpts, p.KernelOptionsAppend...)
 
-	if p.ignition {
-		if p.ignitionPlatform == "" {
-			panic("ignition is enabled but ignition platform ID is not set")
-		}
+	if p.IgnitionPlatform != "" {
 		kernelOpts = append(kernelOpts,
 			"coreos.no_persist_ip", // users cannot add connections as we don't have a live iso, this prevents connections to bleed into the system from the ign initrd
```

Comment:
this looks very specific to a particular caller (Automotive? Edge?)

I think I would pull out `coreos.no_persist_ip` into something else. There is a new `kernel-cmdline.bls-append` stage in https://github.com/osbuild/osbuild/pull/1429

Reply:
This was added by the edge/iot team as part of adding ignition support https://github.com/osbuild/osbuild-composer/pull/3130
@runcom: is this necessary for all ignition-enabled images?

Reply:
I can say at least for CoreOS, we don't want it :)

Reply:
Fixed in #272.